openapi.yaml

```diff
@@ -32299,247 +32355,231 @@ components:
             sentinel_username:
-              description: Sentinel username to authenticate with a Redis Sentinel instance. Requires Redis v6.2.0+.
+              description: 'Sentinel username to authenticate with a Redis Sentinel instance. If undefined, ACL authentication won''t be performed. This requires Redis v6.2.0+.'
               type: string
-              nullable: true
             server_name:
-              description: Server name for SSL verification.
+              description: A string representing an SNI (server name indication) value for TLS.
               type: string
-              nullable: true
             ssl:
               description: 'If set to true, uses SSL to connect to Redis.'
               type: boolean
-              nullable: true
             ssl_verify:
-              description: 'If set to true, verifies the validity of the server SSL certificate.'
+              description: 'If set to true, verifies the validity of the server SSL certificate. If setting this parameter, also configure `lua_ssl_trusted_certificate` in `kong.conf` to specify the CA (or server) certificate used by your Redis server. You may also need to configure `lua_ssl_verify_depth` accordingly.'
               type: boolean
-              nullable: true
             username:
-              description: 'Username to use for Redis connections. If undefined, ACL authentication won''t be performed. Requires Redis v6.0.0+.'
+              description: 'Username to use for Redis connections. If undefined, ACL authentication won''t be performed. This requires Redis v6.0.0+. To be compatible with Redis v5.x.y, you can set it to `default`.'
               type: string
-              nullable: true
-        created_at:
-          description: Unix epoch when the resource was created.
-          type: integer
-          nullable: true
-        id:
-          type: string
-          nullable: true
-        name:
-          type: string
-          nullable: true
-        tags:
-          type: array
-          items:
-            type: string
-          nullable: true
-        type:
-          type: string
-          const: redis-ee
-        updated_at:
-          description: Unix epoch when the resource was last updated.
-          type: integer
-          nullable: true
+      example:
+        config:
+          cluster_nodes:
+            - ip: 192.168.1.10
+              port: 6380
+          connect_timeout: 2000
+          database: 0
+          host: localhost
+          keepalive_pool_size: 256
+          password: password
+          port: 6379
+          read_timeout: 1000
+          send_timeout: 1000
+          sentinel_nodes:
+            - host: sentinel1.redis.server
+              port: 26379
+          server_name: redis-ee
+          ssl: false
+          ssl_verify: false
+          username: username
+        type: redis-ee
       additionalProperties: false
       required:
         - type
-        - config
```

Comment:
`config` is required by the API to create partial - https://developer.konghq.com/api/konnect/control-planes-config/v2/#/operations/create-partial